File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/C/s380356166_commented.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int key;
    int value;
} Pair;

static int read() {
    static char buffer[8192];
    static int pos = 0, read_len = 0;
    if (pos >= read_len) {
        pos = 0;
        read_len = fread(buffer, 1, sizeof(buffer), stdin);
        if (read_len <= 0) {
            exit(EXIT_FAILURE);
        }
    }
    return buffer[pos++];
}

static int next_int() {
    int len = 0;
    char str[4096];
    str[len++] = read();
    while (!isspace(read())) {
        str[len++] = read();
    }
    str[len] = '\0';
    return atoi(str);
}

static long next_long() {
    int len = 0;
    char str[4096];
    str[len++] = read();
    while (!isspace(read())) {
        str[len++] = read();
    }
    str[len] = '\0';
    return atol(str);
}

static char* next_string() {
    int len = 0;
    char str[4096];
    str[len++] = read();
    while (!isspace(read())) {
        str[len++] = read();
    }
    str[len] = '\0';
    return strdup(str);
}

static char next_char() {
    while (1) {
        int c = read();
        if (!isspace(c)) {
            return (char)c;
        }
    }
}

void solve(int test_number, FILE* input, FILE* output) {
    int ans = 0;
    int K = next_int();
    int S = next_int();
    int wk = 0;
    for (int x = 0; x <= K; x++) {
        for (int y = 0; y <= K; y++) {
            wk = S - x - y;
            if (wk >= 0 && wk <= K) {
                ans++;
            }
        }
    }
    fprintf(output, "%d\n", ans);
}

int main() {
    FILE* input = stdin;
    FILE* output = stdout;
    solve(1, input, output);
    fclose(output);
    return 0;
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/Python/s380356166_commented.py
import sys

class Solver:
    def solve(self, test_number, in_stream, out_stream):
        ans = 0
        K = in_stream.nextInt()
        S = in_stream.nextInt()
        for x in range(K + 1):
            for y in range(K + 1):
                wk = S - x - y
                if 0 <= wk <= K:
                    ans += 1
        out_stream.write(f"{ans}\n")

class MyInput:
    def __init__(self, input_stream):
        self.in_stream = input_stream
        self.buffer = []
        self.pos = 0

    def read(self):
        if self.pos >= len(self.buffer):
            self.buffer = self.in_stream.readline().strip()
            self.pos = 0
        if not self.buffer:
            raise EOFError("End of file reached")
        ch = self.buffer[self.pos]
        self.pos += 1
        return ch

    def next_int(self):
        num_str = ""
        while True:
            ch = self.read()
            if ch.isdigit() or ch == '-':
                num_str += ch
            else:
                break
        return int(num_str)

    def next_long(self):
        num_str = ""
        while True:
            ch = self.read()
            if ch.isdigit() or ch == '-':
                num_str += ch
            else:
                break
        return int(num_str)

    def next_string(self):
        string = ""
        while True:
            ch = self.read()
            if not ch.isspace():
                string += ch
            else:
                break
        return string.strip()

def main():
    in_stream = MyInput(sys.stdin)
    out_stream = sys.stdout
    solver = Solver()
    solver.solve(1, in_stream, out_stream)

if __name__ == "__main__":
    main()  # End of Code
